src/main/java/pib/utility/Parser.java

```diff
@@ -0,0 +1,137 @@
+package pib.utility;
+
+import pib.enums.TaskType;
+import pib.pibexception.PibException;
+
+import java.util.Scanner;
+
+/**
+ * Class to parse the user input
+ */
+public class Parser {
+    private Scanner sc;
+    private TaskList list;
+
+    /**
+     * Public constructor to initialise a Parser object
+     *
+     * @param list TaskList that is used to store the Tasks where this parser is used
+     */
+    public Parser(TaskList list) {
+        this.sc = new Scanner(System.in);
+        this.list = list;
+    }
+
+    /**
+     * Uses the scanner to read inputs from the user
+     */
+    public void readInput() {
+        scanner:
+        while (sc.hasNextLine()) {
+            try {
```

Comment:
Any reason why you catch the exceptions in these Parser methods instead of throwing it to the main caller in Pib to catch? Since they all throw the same type of PibException, letting Pib catch these exceptions when it is calling these methods seems like a better control flow to me.

Just a nit from me and I might not be right about this 😃